### ISSUE TITLE
TS.MADD - unsuccessful commands are also being sent to replica (potential cyclic loop)

### ISSUE THREAD

**Issue by ng5** [RedisTimeSeries/RedisTimeSeries#942]
My replica logs are filled with errors that are already caught in the primary instance of redis. If a command has failed in primary, it should never be sent downstream to replica - it is almost always going to fail there too.  this can potentially cause cyclic deadlock incase primary goes down and replica takes over. it will again start cycle if roles are reversed.

1) start two instance of redis: redis1 (primary) and redis2 (replica)

2) In redis1 Run:  TS.MADD TS1 1 1.234  # It is known that TS1 does not already exist because TS.CREATE was never run
 above command fails in primary -  (error) ERR TSDB: the key is not a TSDB key

3) previous command was expected to fail, but it should never result in errors downstream to replicas. however if you tail log file of redis2, you will see below error. 

1:S 14 Dec 2021 22:35:13.456 # == CRITICAL == This replica is sending an error to its master: '-ERR TSDB: the key is not a TSDB key' after processing the command 'ts.madd'
1:S 14 Dec 2021 22:35:13.456 # Latest backlog is: '"\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*1\r\n$4\r\nping\r\n*4\r\n$7\r\nTS.MADD\r\n$3\r\nTS1\r\n$1\r\n1\r\n$5\r\n1.234\r\n*1\r\n$4\r\nping\r\n"'

**Comment by ng5** [comment]
Start Redis 1)
docker run -it --rm --name redis1 --network=host redislabs/redismod --port 26379 --loadmodule /usr/lib/redis/modules/redistimeseries.so DUPLICATE_POLICY LAST

Start Redis 2)
docker run -it --rm --name redis2 --network=host redislabs/redismod --port 26380 --replicaof localhost 26379 --loadmodule /usr/lib/redis/modules/redistimeseries.so DUPLICATE_POLICY LAST

Connect from Redis 1) and Send wrong command and Observe the errors in Redis 2 Replica (Errors being sent downstream)
redis-cli -p 26379

```
127.0.0.1:26379> TS.MADD TS1 1 1.2345
1) (error) ERR TSDB: the key is not a TSDB key
127.0.0.1:26379> 

```


![image](https://user-images.githubusercontent.com/16100870/146092674-1acccaff-fe71-4a84-8ed2-2f8ec323534b.png)

**Comment by danni-m** [comment]
@ng5 thanks for reporting, this was fix in a PR not long ago: https://github.com/RedisTimeSeries/RedisTimeSeries/pull/884

Can you test this on master?

**Comment by ng5** [comment]
@danni-m  thanks for checking out. 
tested with below docker images. (as of 16/12/2021)
1. redislabs/redistimeseries:latest - fixed
2. redislabs/redismod:latest - still not working 
3. redislabs/redismod:edge - still not working

i need to use other modules so prefer to use docker image. hopefully it will be rolled out soon.

**Comment by danni-m** [comment]
@ng5 closing as this was fixed in master and will be released in upcoming 1.6 release
